### PR TITLE
[FIX] l10n_it_edi: Buyer/Seller VAT and country fix

### PR DESCRIPTION
**Pull request by lordkrandel**
Buyer/Seller VAT and country were determined by VAT only. This makes it impossible to determine US TIN correctly, as they were seen just as strings of digits.
We need the stdnum library to rely on the partner's country and anyway verify the VAT for the country we're guessing.

Task: https://www.odoo.com/web#id=3010849&model=project.task
opw-3010849

### CODE REVIEW

File: addons/l10n_it_edi/data/invoice_it_template.xml [hunk 1]
```diff
@@ -58,10 +58,18 @@
...
-                <IdFiscaleIVA>
+                <IdFiscaleIVA t-if="seller.vat and in_eu(seller)">
                     <IdPaese t-esc="get_vat_country(seller.vat)"/>
                     <IdCodice t-esc="get_vat_number(seller.vat)"/>
                 </IdFiscaleIVA>
+                <IdFiscaleIVA t-if="seller.vat and not in_eu(seller)">
+                    <IdPaese t-esc="seller.country_id.code"/>
+                    <IdCodice t-esc="'OO99999999999'"/>
+                </IdFiscaleIVA>
+                <IdFiscaleIVA t-if="not seller.vat and seller.country_id.code != 'IT'">
```
Comment (jco-odoo) [inline addons/l10n_it_edi/data/invoice_it_template.xml:69]: @lordkrandel I wonder if this t-if is a realistic use case, but otherwise lgtm.

Comment (lordkrandel) [reply addons/l10n_it_edi/data/invoice_it_template.xml:69]: Yes, it is, the ticket I talk about in the description is about this.
We also have to investigate this https://www.odoo.com/web#id=3044072&cids=1&menu_id=4720&action=333&active_id=49&model=project.task&view_type=form

Comment (lordkrandel) [reply addons/l10n_it_edi/data/invoice_it_template.xml:69]: Before, wrong
image
After, correct
image

Comment (lordkrandel) [reply addons/l10n_it_edi/data/invoice_it_template.xml:69]: I've made a simple vendor bill with a US partner (GMass, which is Google)

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 2]
```diff
@@ -117,8 +117,8 @@ def _l10n_it_edi_check_ordinary_invoice_configuration(self, invoice):
             errors.append(_("Tax representative partner %s of %s must have a tax number.", seller.l10n_it_tax_representative_partner_id.display_name, seller.display_name))
 
         # <1.4.1>
-        if not buyer.vat and not buyer.l10n_it_codice_fiscale and buyer.country_id.code == 'IT':
-            errors.append(_("The buyer, %s, or his company must have a VAT number and/or a tax code (Codice Fiscale).", buyer.display_name))
+        if not buyer.vat and not (buyer.l10n_it_codice_fiscale and buyer.country_id.code == 'IT'):
+            errors.append(_("The buyer, %s, or his company must have a VAT number and/or an Italian tax code (Codice Fiscale).", buyer.display_name))
```
Comment (jco-odoo) [inline addons/l10n_it_edi/models/account_edi_format.py:121]: @lordkrandel Not sure why this change is here.  This only says that if the buyer is from Italy you should at least have a codice fiscale or a vat and you make it like as a foreigner you also need a vat.

**Comment by jco-odoo** [comment]
@TonyMasciI

**Comment by TonyMasciI** [comment]
> @TonyMasciI

Why should a US company create an customer electronic invoice according to the Italian standard? If it has a USA's VAT? I can't read the task (https://www.odoo.com/web#id=3010849&model=project.task). 

The problem is import of electronic vendor bill or export of the customer invoice?

As is it seems that you want to use italian eletronic invoice for foreign company, that obviusly dasen't have italian VAT.

**Comment by lordkrandel** [comment]
@TonyMasciI

See: https://fiscomania.com/operazioni-estere-autofattura-integrazione/#autofattura
or: https://aranciadigitale.it/fattura-elettronica/fatture-lestero-cosa-bisogna-sapere/ (see Autofatture)

If a US vendor sells you services, he will send you an invoice with no taxes (it's an export).
As you only pay taxes based on where you buy, it's your duty as a buyer to integrate it by paying taxes to the Italian Tax Agency.
The US vendor doesn't know which Italian taxes will be applied to his services, and that's normal.

The Agenzia delle Entrate also requires that you send a "self-invoice" (Autofattura) where you also specify the taxes details of this vendor bill. So this use case is an **export** to the SDI of a "self-invoice" generated **from a Vendor Bill of a foreign vendor**.
In this self invoice export, the buyer must be your company, the seller must be your vendor. (it's reverse, that's why it's called Autofattura)

File: addons/l10n_it_edi/models/account_edi_format.py (hunk 2 above)
Comment (lordkrandel) [reply addons/l10n_it_edi/models/account_edi_format.py]: Removed.

**Comment by TonyMasciI** [comment]
> @TonyMasciI

@lordkrandel 

Everything is clear for us, we did a test with an XML file, it's ok.

**Comment by jco-odoo** [comment]
I am still having doubts about the 00000 case because we have a validation to have the seller's vat to be required.

**Comment by TonyMasciI** [comment]
> I am still having doubts about the 00000 case because we have a validation to have the seller's vat to be required.

We have tested this case, it is correct

**Comment by jco-odoo** [comment]
@TonyMasciI It is just that we need to allow it then.

**Comment by lordkrandel** [comment]
We are guessing that the right check is incorrect.
image
Should be like
image

**Comment by jco-odoo** [comment]
@lordkrandel The elif should be a separate if: if len(seller.vat or '') > 30:

**Comment by TonyMasciI** [comment]
> @lordkrandel The elif should be a separate if: if len(seller.vat or '') > 30:

Why with separate "if"? Only if company have VAT you have to check the lenght of VAT, not in other case. But you can add other check for "seller".

**Comment by TonyMasciI** [comment]
> format_vat

Ok!

**Review by TonyMasciI — APPROVED** [review]

**Comment by jco-odoo** [comment]
@robodoo r+
